hashicorp/waypoint
# ui: redirect from `:id` to `seq/:seq` for all detail routes

## Why the change?

This change was deferred from #1513 and #1863. The immediate motivation is to get rid of some irksome TypeScript errors. This turned out to be the cleanest and quickest way to achieve that goal. I think it’s a better UX overall that the app treats the sequence-based URLs as canonical.

## What does it look like?

### Before

https://user-images.githubusercontent.com/34030/127306754-5deef9d6-5cdc-48ba-956c-7d273e98c256.mp4

### After

https://user-images.githubusercontent.com/34030/127306798-12cce88d-199e-4af0-a68a-31eccd1c97f3.mp4

## How do I test it?

1. Check out the branch `get checkout ui/id-route-refactors`
2. Boot the dev server `cd ui && yarn start`
3. [Visit a build by sequence number](http://localhost:4200/default/marketing-public/app/wp-matrix/build/seq/4) (e.g. `.../build/seq/4`)
4. Verify the build details are rendered
5. [Visit the same build by ID](http://localhost:4200/default/marketing-public/app/wp-matrix/build/RMZNQAAGS732GGW3AWA44I03) (e.g. `.../build/RMZNQAAGS732GGW3AWA44I03`)
6. Verify you see `.../build/seq/4` in the URL bar
7. [Visit a deployment by sequence number](http://localhost:4200/default/marketing-public/app/wp-matrix/deployment/seq/4) (e.g. `.../deployment/seq/4`)
8. Verify the deployment details are rendered
9. [Visit the same deployment by ID](http://localhost:4200/default/marketing-public/app/wp-matrix/deployment/TC0G63BI4VTJCAXCKWV8U0WY) (e.g. `.../deployment/TC0G63BI4VTJCAXCKWV8U0WY`)
10. Verify you see `.../deployment/seq/4` in the URL bar
11. [Visit a release by sequence number](http://localhost:4200/default/marketing-public/app/wp-matrix/release/seq/3) (e.g. `.../release/seq/3`)
12. Verify the release details are rendered
13. [Visit the same release by ID](http://localhost:4200/default/marketing-public/app/wp-matrix/deployment/57MOJ5NI5ZR571IWSP0BBEV2) (e.g. `.../release/57MOJ5NI5ZR571IWSP0BBEV2`)
14. Verify you see `.../release/seq/3` in the URL bar
